jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -183,12 +191,15 @@ public String toString()
             return _string;
         }
 
+        /**
+         * @return {@code true} If the {@link Charset} for this type is assumed rather than being explicitly declared.
+         */
         public boolean isCharsetAssumed()
         {
             return _assumedCharset;
         }
 
-        public HttpField getContentTypeField()
+        public ContentTypeField getContentTypeField()
```

Comment:
Should not the method below `public HttpField getContentTypeField(Charset charset)` also have its return type changed?

Reply:
Not really, because there are two aspect of `ContentTypeField`: it has a `getMimeType` method and it is a `PreEncodedHttpField`.   For an arbitrary contentType returned from this method, it is not guaranteed that it will have a known mimeType and it is not known if it is worthwhile going to the effort of pre-encoding.

That is why I think this class belongs as a subtype of this class, as it is pretty special purpose for long held constants for known types.